### PR TITLE
api: Add path restrictions in object api

### PR DESCRIPTION
**Pull request by r-rojo**
Add checks for object names that start with `.` or `/` and disallow if
found.
Add unit tests for these checks

### CODE REVIEW

File: libopae/plugins/xfpga/sysobject.c [hunk 1]
```diff
@@ -47,6 +47,13 @@
 		}                                                              \
 	} while (false);
 
+#define VALIDATE_NAME(_N)                                                      \
+	do {                                                                   \
+		if (_N[0] == '.' || _N[0] == '/') {                            \
```
Comment (tswhison) [inline libopae/plugins/xfpga/sysobject.c]: What if .. appears, but not as the first character, eg "errors/../../../../../../../../../../../"?

File: tools/extra/ras/main.c [hunk 2]
```diff
@@ -1064,26 +1064,24 @@ fpga_result mmio_error(fpga_handle afu_handle, struct RASCommandLine *rasCmdLine
 	if ( rasCmdLine->function >0 )
 		function = rasCmdLine->bus;
 
-	result = fpgaHandleGetObject(afu_handle, FPAG_DEVICEID_PATH, &dev_object, 0);
+	result = fpgaGetPropertiesFromHandle(afu_handle, &props);
 	if (result != FPGA_OK) {
-		OPAE_ERR("Failed to get Handle Object");
+		OPAE_ERR("Failed to get properteis from handle");
```
Comment (tswhison) [inline tools/extra/ras/main.c]: s/properteis/properties/

**Comment by zhangzhang10** [comment]
@r-rojo @tswhison Do we need to check if a sysobject name may be a symbolic link? This is a required SDL check, but I'm not sure if we should do it here.

**Comment by tswhison** [comment]
@zhangzhang10 Since we're dealing with sysfs hear, the likelihood of encountering a symlink is very high; however, I think we are ok in that the path is always relative to the root stored in the token.  I believe that it is impossible to create an arbitrary rogue symlink in sysfs (at least without a driver).

**Comment by zhangzhang10** [comment]
@tswhison I agree with what you said. Thanks for the explanation.

**Pull request merged**